src/lib/validators.js

```diff
@@ -132,3 +133,26 @@ export const validateUnique = arr => {
 
 export const validateChild = ourShip => ship =>
   ourShip !== ob.sein(ship) && `This point is not a child of ${ourShip}.`;
+
+export const validatePsbt = base64 => {
+  try {
+    bitcoin.Psbt.fromBase64(base64);
+  } catch (e) {
+    return 'Invalid Partially Signed Bitcoin Transaction';
+  }
+};
+
+export const validateSignablePsbt = hd => base64 => {
+    const psbt = bitcoin.Psbt.fromBase64(base64);
+    return !psbt.data.inputs.some((input, idx) => {
+      try {
+        //  removing already derived part, eg 'm/84'/0'/0'/0/0' becomes '0/0'
+        const path = input.bip32Derivation[0].path.substring(12);
```

Comment:
Is the path here a constant or something? Seems a tad dangerous to assume things about the path's prefix/length.

(Also, in the comment, probably don't bother with the enclosing single quotes. It's kind of confusing to read in this context lol.)

Reply:
Rofl at the single quotes, fixed.

The paths in the transaction generated in urbit will always begin with `m/84'/0'/0'`, and if not, this will just fail to sign.

`btcWallet` is derived with `m/84'/0'/0'`, so when we look at the input derivation path of the PSBT we have to derive the remaining part.